File: Super-Admin/backend/models/school.js
const mongoose = require("mongoose");

const School = mongoose.model(
  "school",
  new mongoose.Schema({
    code: {
      type: String,
      trim: true,
      required: true,
      unique: true,
    },
    name: {
      type: String,
      trim: true,
      required: true,
    },
    principal: {
      type: String,
      trim: true,
      required: true,
    },
    email: {
      type: String,
      trim: true,
      required: true,
      unique: true,
    },
    website: {
      type: String,
      trim: true,
    },
    state: {
      type: String,
      trim: true,
      required: true,
    },
    city: {
      type: String,
      trim: true,
      required: true,
    },
    address: {
      type: String,
      trim: true,
      required: true,
    },
    contact: {
      type: [Number],
      trim: true,
      required: true,
    },
    doi: {
      type: Date,
      required: true,
    },
    joined: {
      type: Date,
      default: Date.now(),
    },
    active: {
      type: Boolean,
      required: true,
    },
  })
);

module.exports = School;

File: Super-Admin/bentzip/lib/models/school.dart
class School{
  String? name;
  String? email;
  String? doi;
  String? id;
  String? contact;
  String? principal;
  String? website;
  String? state;
  String? city;
  String? address;
  bool? active;

  Map<String, dynamic> toJson(){
     return {
       "_id":id,
       "name":name,
       "principal":principal,
       "email":email,
       "state":state,
       "city":city,
       "address":address,
       "contact":List.of({contact}),
       "doi":doi.toString(),
       "active":active
     };
   }

}